saleor/plugins/webhook/shipping.py

```diff
@@ -0,0 +1,161 @@
+import base64
+import logging
+from collections import defaultdict
+from typing import TYPE_CHECKING, Any, Callable, Dict, List
+
+import graphene
+from django.core.cache import cache
+from django.db.models import QuerySet
+from prices import Money
+
+from ...shipping.interface import ShippingMethodData
+from ..base_plugin import ExcludedShippingMethod
+from .const import CACHE_EXCLUDED_SHIPPING_TIME, EXCLUDED_SHIPPING_REQUEST_TIMEOUT
+from .tasks import _get_webhooks_for_event, trigger_webhook_sync
+from .utils import APP_ID_PREFIX
+
+if TYPE_CHECKING:
+    from ...app.models import App
+
+
+logger = logging.getLogger(__name__)
+
+
+def to_shipping_app_id(app: "App", shipping_method_id: str) -> "str":
+    return base64.b64encode(
+        str.encode(f"{APP_ID_PREFIX}:{app.pk}:{shipping_method_id}")
+    ).decode("utf-8")
+
+
+def parse_list_shipping_methods_response(
+    response_data: Any, app: "App"
+) -> List["ShippingMethodData"]:
+    shipping_methods = []
+    for shipping_method_data in response_data:
+        method_id = shipping_method_data.get("id")
+        method_name = shipping_method_data.get("name")
+        method_amount = shipping_method_data.get("amount")
+        method_currency = shipping_method_data.get("currency")
+        method_maximum_delivery_days = shipping_method_data.get("maximum_delivery_days")
+
+        shipping_methods.append(
+            ShippingMethodData(
+                id=to_shipping_app_id(app, method_id),
+                name=method_name,
```

Comment:
In case when we have an `App` with > 2 methods without 'id' (reading the code it's possible because of `.get()`) result from `to_shipping_app_id` is not unique. Consider use `enumerate` and return something unique in `get()` instead of `None`.

Reply:
That's not a new piece of code, but rather me moving it between files to avoid import spaghetti. I'd prefer not to change the implementation without a dedicated task.